aria-practices.html

```diff
@@ -6338,6 +6338,127 @@ <h3>Indicating sort order with <code>aria-sort</code></h3>
+      <pre><code>&lt;svg width="1000" height="1000"&gt;
+  &lt;text x="10" y="10" role="heading" aria-level="1"&gt;Total Trees:&lt;/text&gt;
+  &lt;text x="10" y="100" role="heading" aria-level="2"&gt;Deciduous Trees:&lt;/text&gt;
+  &lt;text x="10" y="200" role="heading" aria-level="2"&gt;Evergreen Trees:&lt;/text&gt;
+  ...
+&lt;/svg&gt;
+</code></pre>
+
+      <section id="aria-level_heading_role_remediation">
+        <h4>Remediation Uses Cases for <code>aria-level</code></h4>
+        <p>In rare scenarios, legacy code cannot be converted to using HTML heading elements. In these cases, the <code>aria-level</code> attribute and the <code>heading</code> role can be added to the web site's heading elements to communicate to assistive technologies that those elements should be treated like a native HTML section heading element.</p>
+        <p>For example, "Definition of a Room" is a subsection of "Housing Specification":</p>
+        <pre><code>&lt;div role="heading" aria-level="1" class="heading-big"&gt;Housing Specification&lt;/div&gt;
+&lt;div role="heading" aria-level="2" id="room-definition" class="heading-small"&gt;Definition of a Room&lt;/div&gt;
+...
+</code></pre>
+        <p>Equivalent (and preferred) HTML:</p>
+        <pre><code>&lt;h1&gt;Housing Specification&lt;/h1&gt;
+&lt;h2 id="room-definition"&gt;Definition of a Room&lt;/h2&gt;
+...
+</code></pre>
+      </section>
+    </section>
+
+    <section id="aria-level_listitem_role">
+      <h3>Using <code>aria-level</code> for the <code>listitem</code> Role</h3>
+      <p>
+        The <code>aria-level</code> attribute can be used on elements with <code>listitem</code> role to represent list subitems when the structure of the DOM tree does not imply the intended level of nesting.
+        This is a rare scenario, however.
+        For example, consider a subset of a Table of Contents that only covers the current chapter,
+        and omits the top-level item (the chapter).
+        The first list item is semantically a second-level item.
+        This can be indicated using <code>aria-level</code>, as follows:
+      </p>
+      <pre><code>&lt;h1>Preface&lt;/h1>
+&lt;h2>Table of Contents&lt;/h2>
+&lt;ul>
+  &lt;li aria-level="2">&lt;a href="#intended-audience">Intended audience&lt;/a>&lt;/li>
+  ...
+&lt;/ul>
+&lt;h2 id="intended-audience">Intended audience&lt;/h2>
+...
+</code></pre>
+
+      <p class="note">The HTML <code>li</code> element uses the <code>listitem</code> role by default.</p>
+    </section>
+
+    <section id="aria-level_treeitem_role">
+      <h3>Using <code>aria-level</code> for the <code>treeitem</code> Role</h3>
+      <p>In some implementations of the <a href="#TreeView">Tree View Design Pattern</a>, authors might need to explicitly declare the <code>aria-level</code> attribute.</p>
```

Comment:
It feels to me like there needs to be one more sentence (in its own paragraph) before this one, that describes "normal" use of aria-level for the treeitem role. It feels like we've jumped into the exceptional case, and didn't even mention the typical case.

Reply:
@zcorpan 
Oh. I guess aria-level is normally computed by the browser, based on DOM structure.

It would still be nice to have a short intro sentence like, "Usually, aria-level is computed by the browser based on DOM structure.", then follow with "In some implementations...".

Reply:
@mcking65 clarified this in a716ad878458eadeaf28e6a82b59995b75de1d55